model/remote_cluster.go

```diff
@@ -35,48 +41,219 @@ func (rc *RemoteCluster) PreSave() {
 	if rc.CreateAt == 0 {
 		rc.CreateAt = GetMillis()
 	}
+	rc.fixTopics()
+}
+
+func (rc *RemoteCluster) IsValid() *AppError {
+	if !IsValidId(rc.RemoteId) {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.id.app_error", nil, "id="+rc.RemoteId, http.StatusBadRequest)
+	}
+
+	if rc.DisplayName == "" {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.name.app_error", nil, "display_name empty", http.StatusBadRequest)
+	}
 
-	if rc.LastPingAt == 0 {
-		rc.LastPingAt = rc.CreateAt
+	if rc.CreateAt == 0 {
+		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.create_at.app_error", nil, "create_at=0", http.StatusBadRequest)
 	}
+	return nil
 }
 
-func (rc *RemoteCluster) IsValid() *AppError {
-	if !IsValidId(rc.Id) {
-		return NewAppError("RemoteCluster.IsValid", "model.cluster.is_valid.id.app_error", nil, "id="+rc.Id, http.StatusBadRequest)
+func (rc *RemoteCluster) PreUpdate() {
+	rc.fixTopics()
+}
+
+// fixTopics ensures all topics are separated by one, and only one, space.
+func (rc *RemoteCluster) fixTopics() {
+	trimmed := strings.TrimSpace(rc.Topics)
+	if trimmed == "" {
+		rc.Topics = ""
+		return
+	}
+	if trimmed == "*" {
+		rc.Topics = "*"
+		return
```

Comment:
What about the condensed form:

```go
if trimmed == "" || trimmed == "*" {
  rc.Topics = trimmed
  return
}
```

Reply:
Fixed via https://github.com/mattermost/mattermost-server/pull/16432/commits/c785b4b692a65411487822c2ce7c40f614bdd121